### PR TITLE
PP-12302 Refactor integration tests to use JUnit5

### PR DESCRIPTION
**Pull request by DomBelcher**
## WHAT YOU DID
Refactor RefundReversalResourceIT integration test to use JUnit5
Add constructor to AppWithPostgresAndSqsExtension accepting custom extension of ConnectorApp class

## How to test
Run integration tests

### CODE REVIEW

File: src/test/java/uk/gov/pay/connector/it/resources/stripe/StripeRefundsResourceIT.java [hunk 1]
```diff
@@ -38,13 +38,13 @@ public class StripeRefundsResourceIT {
     private String stripeAccountId = "stripe_account_id";
     private String accountId = String.valueOf(nextLong());
 
-    private StripeMockClient stripeMockClient;
+//    private StripeMockClient stripeMockClient;
```
Comment (SandorArpa) [inline src/test/java/uk/gov/pay/connector/it/resources/stripe/StripeRefundsResourceIT.java]: Can we remove commented out code?

File: src/test/java/uk/gov/pay/connector/it/resources/stripe/StripeRefundsResourceIT.java [hunk 2]
```diff
@@ -38,13 +38,13 @@ public class StripeRefundsResourceIT {
...
     private DatabaseFixtures.TestAccount defaultTestAccount;
     private DatabaseFixtures.TestCharge defaultTestCharge;
     
     @BeforeEach
     void setUpStripe() {
-        stripeMockClient = new StripeMockClient(app.getWiremockserver());
+//        stripeMockClient = new StripeMockClient(app.getWiremockserver());
```
Comment (SandorArpa) [inline src/test/java/uk/gov/pay/connector/it/resources/stripe/StripeRefundsResourceIT.java]: Same as above

**Review by SandorArpa — CHANGES_REQUESTED** [review]

File: src/test/java/uk/gov/pay/connector/it/resources/stripe/StripeRefundsResourceIT.java (hunk 1 above)
Comment (DomBelcher) [reply src/test/java/uk/gov/pay/connector/it/resources/stripe/StripeRefundsResourceIT.java]: Fixed

File: src/test/java/uk/gov/pay/connector/it/resources/stripe/StripeRefundsResourceIT.java (hunk 2 above)
Comment (DomBelcher) [reply src/test/java/uk/gov/pay/connector/it/resources/stripe/StripeRefundsResourceIT.java]: Fixed

**Review by SandorArpa — APPROVED** [review]
🍬

**Pull request merged**